src/modules/powerrename/lib/Settings.cpp

```diff
@@ -470,12 +359,205 @@ void CRenameMRU::_FreeMRUList()
+    return settings.persistState;
+}
+
+void CSettings::SetPersistState(bool persistState)
+{
+    std::scoped_lock lock{ dataLock };
+    settings.persistState = persistState;
+}
+
+bool CSettings::GetMRUEnabled() const
+{
+    std::scoped_lock lock{ dataLock };
+    return settings.MRUEnabled;
+}
+
+void CSettings::SetMRUEnabled(bool MRUEnabled)
+{
+    std::scoped_lock lock{ dataLock };
+    settings.MRUEnabled = MRUEnabled;
+}
+
+long CSettings::GetMaxMRUSize() const
+{
+    std::scoped_lock lock{ dataLock };
+    return settings.maxMRUSize;
+}
+
+void CSettings::SetMaxMRUSize(long maxMRUSize)
+{
+    std::scoped_lock lock{ dataLock };
+    settings.maxMRUSize = maxMRUSize;
+}
+
+long CSettings::GetFlags() const
+{
+    std::scoped_lock lock{ dataLock };
+    return settings.flags;
+}
+
+void CSettings::SetFlags(long flags)
+{
+    std::scoped_lock lock{ dataLock };
+    settings.flags = flags;
+}
+
+const std::wstring& CSettings::GetSearchText() const
+{
+    std::scoped_lock lock{ dataLock };
+    return settings.searchText;
+}
+
+void CSettings::SetSearchText(const std::wstring& text)
+{
+    std::scoped_lock lock{ dataLock };
+    settings.searchText = text;
+}
+
+const std::wstring& CSettings::GetReplaceText() const
+{
+    std::scoped_lock lock{ dataLock };
+    return settings.replaceText;
+}
+
+void CSettings::SetReplaceText(const std::wstring& text)
+{
+    std::scoped_lock lock{ dataLock };
+    settings.replaceText = text;
+}
+
+void CSettings::LoadPowerRenameData()
+{
+    std::scoped_lock lock{ dataLock };
+
+    if (!std::filesystem::exists(jsonFilePath))
+    {
+        MigrateSettingsFromRegistry();
+    }
+    else
+    {
+        ParseJsonSettings(GetPersistPowerRenameJSON());
+
+        SavePowerRenameData();
+    }
+}
+
+void CSettings::SavePowerRenameData() const
+{
+    std::scoped_lock lock{ dataLock };
+
+    json::JsonObject jsonData;
+
+    jsonData.SetNamedValue(c_enabled,                 json::value(settings.enabled));
+    jsonData.SetNamedValue(c_showIconOnMenu,          json::value(settings.showIconOnMenu));
+    jsonData.SetNamedValue(c_extendedContextMenuOnly, json::value(settings.extendedContextMenuOnly));
+    jsonData.SetNamedValue(c_persistState,            json::value(settings.persistState));
+    jsonData.SetNamedValue(c_mruEnabled,              json::value(settings.MRUEnabled));
+    jsonData.SetNamedValue(c_maxMRUSize,              json::value(settings.maxMRUSize));
+    jsonData.SetNamedValue(c_flags,                   json::value(settings.flags));
+    jsonData.SetNamedValue(c_searchText,              json::value(settings.searchText));
+    jsonData.SetNamedValue(c_replaceText,             json::value(settings.replaceText));
+
+    json::to_file(jsonFilePath, jsonData);
+}
+
+json::JsonObject CSettings::GetPersistPowerRenameJSON()
+{
+    std::scoped_lock lock{ dataLock };
+
+    auto jsonData = json::from_file(jsonFilePath);
+    return jsonData ? jsonData.value() : json::JsonObject();

```

Comment:
there's no point in types if we convert everything in a string. please return `std::optional` and don't call `ParseJsonSettings`, since you already provide sane default values.